core/shared/src/main/scala/zio/clock/package.scala

```diff
@@ -27,18 +27,41 @@ package object clock {
 
   object Clock extends PlatformSpecific with Serializable {
     trait Service extends Serializable {
+
       def currentTime(unit: TimeUnit): UIO[Long]
+
+      // Could be UIO. We keep IO to preserve binary compatibility.
       def currentDateTime: IO[DateTimeException, OffsetDateTime]
-      def instant: UIO[java.time.Instant]                               = currentTime(TimeUnit.MILLISECONDS).map(java.time.Instant.ofEpochMilli(_))
+
+      // The implementation is only here to preserve binary compatibility.
+      def instant: UIO[java.time.Instant] = currentTime(TimeUnit.MILLISECONDS).map(java.time.Instant.ofEpochMilli(_))
```

Comment:
Maybe keep these abstract and provide helper methods for default implementations like we do in other services? I think providing default implementations potentially results in counterintuitive behavior with like mock implementations.

Reply:
I'd also prefer to do that, but it breaks binary compatibility. If we decide that's OK here, I'll remove the default implementation.